### PR TITLE
DM-39698: Add time-based replotting

### PR DESCRIPTION
**Pull request by mfisherlevine**

### CODE REVIEW

File: python/lsst/rubintv/production/slac/botTesting.py
Comment (taranu) [inline python/lsst/rubintv/production/slac/botTesting.py:400]: Not that it really matters, but just FYI you can do `.split('_', 1)`.

File: python/lsst/rubintv/production/slac/botTesting.py
Comment (taranu) [inline python/lsst/rubintv/production/slac/botTesting.py:402]: Why not merge this into one `if self.instrument == 'LSST-TS8' or self.instrument == 'LSSTCam':` statement, if they're identical?

File: python/lsst/rubintv/production/slac/botTesting.py
Comment (taranu) [inline python/lsst/rubintv/production/slac/botTesting.py:685]: Missing docstring for timeout.

File: python/lsst/rubintv/production/slac/botTesting.py
Comment (taranu) [inline python/lsst/rubintv/production/slac/botTesting.py:769]: This is where I say this should definitely be a (data)class.

File: python/lsst/rubintv/production/utils.py
Comment (taranu) [inline python/lsst/rubintv/production/utils.py:795]: I generally prefer having lengthier conditions in parentheses like `(time.time() - start < timeout)` for clarity, even if it's not necessary. YMMV.

**Review by taranu — APPROVED** [review]
Seems fine to me. Take the comments/suggestions as you will.

File: python/lsst/rubintv/production/slac/botTesting.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/slac/botTesting.py:402]: They're actually not quite identical - different detector lists for the different instruments. Going to merge this as-is here, but just FYI that all this stuff will get ripped out and done properly in DM-39922

File: python/lsst/rubintv/production/slac/botTesting.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/slac/botTesting.py]: Yup, definitely agree. Could you just take a quick look at `c7882a5` and check that's what you meant? I think that making the items `Callable` and making the call to `partial()` in making the dataclasses is the right way to go, so that the calling code is basically unchanged, as opposed to putting the args and kwargs in the dataclass, but just wanted to check this is what you were thinking.

File: python/lsst/rubintv/production/utils.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/utils.py:795]: Totally agree 👍

**Pull request merged**
